include/arch/arm/arch/32/mode/object/structures.bf

```diff
@@ -102,6 +102,22 @@ block vcpu_cap {
 }
 #endif
 
+#if CONFIG_MAX_NUM_NODES == 1
```

Comment:
Why don't we check `CONFIG_SMP_ENABLED` here and in the other places? Or add a dedicated feature define for this, so using  `CONFIG_MAX_NUM_NODES` directly is avoided. On RISC-V I have been running into some use cases now where having certain multi core feature available even with just one seL4 core comes handy and the checks for `CONFIG_MAX_NUM_NODES` are a pain then.